packages/components/src/query-controls/author-select.tsx

```diff
@@ -32,6 +32,7 @@ export default function AuthorSelect( {
 					? String( selectedAuthorId )
 					: undefined
 			}
+			__nextHasNoMarginBottom
```

Comment:
Looks like this won't quite work outside of the block inspector. Could you check in the Storybook to see if it actually needs a VStack?

<img width="270" alt="Uneven spacing in the Storybook instance for QueryControls" src="https://user-images.githubusercontent.com/555336/216066290-98c48afd-b129-409e-aa89-3af697ea2f57.png">

Reply:
If we add VStack, then a couple of things come up:

1. In Storybook, all looks good except for the help text. Is it worth adding `StyledHelp` to `FormTokenField`? It already has `StyledLabel`. I would do that in a separate PR if this makes sense. 
2. In the editor, while looking at the Latest Posts block, the spacing is too large with VStack (due to the inspector's 24px margin bottom styles). It would need `spacing="0",` but then QueryControls in Storybook would end up looking the same as your screenshot above. We could add a CSS override to the Latest Posts block but then I believe we'd need to add a CSS class to QueryControls? 

@mirka thoughts? 

**QueryControls in Storybook:**

| Trunk  | With VStack |
| ------------- | ------------- |
| <img width="243" alt="Screenshot 2023-02-06 at 7 18 41 PM" src="https://user-images.githubusercontent.com/35543432/217140539-8abf3211-7f75-4fd1-94c2-e62bf3781776.png">  | <img width="241" alt="Screenshot 2023-02-06 at 7 21 50 PM" src="https://user-images.githubusercontent.com/35543432/217140576-06d7d65d-1379-43d9-8b3d-b248c4171bef.png"> |

**QueryControls in Latest Posts block:**

| Trunk  | With VStack |
| ------------- | ------------- |
| <img width="281" alt="Screenshot 2023-02-06 at 7 14 59 PM" src="https://user-images.githubusercontent.com/35543432/217140842-483f1c6f-ad2a-46a7-b703-a0e65b4d9862.png"> | <img width="278" alt="Screenshot 2023-02-06 at 7 14 49 PM" src="https://user-images.githubusercontent.com/35543432/217140837-8b8ccb8b-c943-44d5-97dd-926018600666.png"> |

Reply:
> 1. In Storybook, all looks good except for the help text. Is it worth adding `StyledHelp` to `FormTokenField`? It already has `StyledLabel`. I would do that in a separate PR if this makes sense.

Sure! FormTokenField actually needs to have its margin deprecated (part of #39358). Migrating to `StyledHelp` would be part of that.

> 2. In the editor, while looking at the Latest Posts block, the spacing is too large with VStack (due to the inspector's 24px margin bottom styles). It would need `spacing="0",` but then QueryControls in Storybook would end up looking the same as your screenshot above. We could add a CSS override to the Latest Posts block but then I believe we'd need to add a CSS class to QueryControls?

QueryControls itself should only care about looking correct in isolation (Storybook), not in the block editor. The block editor needs to counteract any of its own margin hacks on its own. That would likely be in [this file](https://github.com/WordPress/gutenberg/blob/805618c9ee1f45eb0634a723ad903b87fb19d992/packages/block-editor/src/components/block-inspector/style.scss#L23), like we recently did for FocalPointPicker. I _think_ we wouldn't need to add new CSS classes to FormTokenField itself, but let me know if that's not the case.

Reply:
>QueryControls itself should only care about looking correct in isolation (Storybook), not in the block editor. 

Thank you! This is a good reminder. ❤️ 

> I think we wouldn't need to add new CSS classes to FormTokenField itself, but let me know if that's not the case.
Sorry, I meant the Latest Posts block. 

I've updated `QueryControls` so it looks better in Storybook. The spacing should still be adjusted for the help below `FormTokenField` but I don't think it's too bad. And it currently looks better than it does in trunk, due to the lack of even spacing above categories. So I figured it may be better to get this out sooner than later.